Controller/Notification/Index.php

```diff
@@ -103,31 +105,39 @@ public function __construct(
     }
 
     /**
+     * @throws InputException
      * @throws InvalidExternalIdException
-     * @throws InvalidSignatureException
+     * @throws NoSuchEntityException
      */
     public function execute()
     {
         $jsonContent = $this->request->getContent();
         $content = json_decode($jsonContent, true);
-        $application = $this->getApplication($content['externalId'])
-                            ->getData();
 
-        if (empty($application)) {
-            throw new InvalidExternalIdException('Invalid external ID!');
-        }
+        if (isset($content['externalId'])) {
+            $application = $this->getApplication($content['externalId'])
+                ->getData();
 
-        if (!$this->isValidSignature($jsonContent)) {
-            $this->getResponse()
-                ->setStatusCode(400)
-                ->setContent('Failed comparission of CR-Signature and shop hash.');
+            if (empty($application)) {
+                throw new InvalidExternalIdException('Invalid external ID!');
```

Comment:
Czy pustość wniosku jest błędem TYLKO w przypadku, gdy nie jest podany `externalId`?

Reply:
W tym wypadku tak, bo brak statusu nic nie zmienia we flow

Reply:
Ja pytałem o `externalId`. Faktycznie przy niepodaniu `externalId` błędu nie ma, a jeśli jest podany, a nie ma wniosku, to błąd już jest? Pytam, bo konkretnie na `externalId` powstał `if`.

Reply:
Jak externalId nie ma to nie ma co robic

Reply:
I zapytanie bez `externalId` jest normalne i nie powinno wywoływać błędu. Powinno być oznaczone jako sukces dostarczenia statusu i odpowiedzieć `200 OK`. Tak?

Reply:
Powinno być jako 404

Reply:
Chociaż jeśli notyfikacja poleci zła i 404 bedzie to nadal będzie ponawiał